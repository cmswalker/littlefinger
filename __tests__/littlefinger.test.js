const littlefinger = require('../lib');

const CWD = process.cwd();

const dependencies = {
  eslint:
    'https://raw.githubusercontent.com/cmswalker/littlefinger/master/fixtures/remotes-eslint.json',
  functional:
    'https://raw.githubusercontent.com/cmswalker/littlefinger/master/fixtures/remotes-functional.json',
  http:
    'https://raw.githubusercontent.com/cmswalker/littlefinger/master/fixtures/remotes-http.json'
};

const awaitBuild = config => littlefinger.build({ ...config });

describe('littlefinger', () => {

  let types;
  let output;

  beforeEach(() => {
    types = { dependencies: true, devDependencies: true, peerDependencies: true };
    output = `${CWD}/__tests__/fixtures/remote-test.json`;
  });

  describe('happy path', () => {
    it('should work as described', async () => {
      await expect(awaitBuild({ output, dependencies, types })).resolves.toMatchSnapshot();
    });
  });

  describe('error path', () => {
    it('should require types', async () => {
      // TODO: assert actual error messages per step
      await expect(awaitBuild({ dependencies: {} })).rejects.toThrow();
    });
  });
});
