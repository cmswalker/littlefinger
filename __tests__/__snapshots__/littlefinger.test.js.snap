// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`littlefinger happy path should work as described 1`] = `
Object {
  "dependencies": Object {
    "ramda": "^0.25.0",
    "superagent": "^3.8.2",
  },
  "devDependencies": Object {
    "eslint": "^4.15.0",
    "eslint-config-airbnb-base": "^12.1.0",
    "eslint-plugin-import": "^2.8.0",
  },
  "peerDependencies": Object {},
}
`;
